File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
  if (input == input.toUpperCase() && input.match(/[a-z]/i)) {
    return "Whoa, chill out!";
  } else if (input[input.length-1] == "?" ){
    return "Sure."
  } else if (input.trim().length == 0) {
    return "Fine. Be that way!";
  } else {
    return "Whatever.";
  }
};

module.exports = Bob;

File: scala/bob/src/main/scala/bob.scala
class Bob {
  def hey(msg: String) : String = {
    var response = ""
    if(msg.map(_.toUpper) == msg && msg.matches("^.*[a-zA-Z].*$")) {
      response = "Woah, chill out!"
    }else if(msg.matches(".*\\?")) {
      response = "Sure."
    }else if(msg.trim == "") {
      response = "Fine. Be that way!"
    }else{
      response = "Whatever."
    }
    return response
  }
}
